source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -69,6 +114,18 @@ void InstanceImpl::ThreadLocalPool::onClusterAddOrUpdateNonVirtual(
   }
 
   ASSERT(cluster_ == nullptr);
+  if (cluster.info()->clusterType().name() == Extensions::Clusters::ClusterTypes::get().Redis) {
+    auto clusters = parent_.cm_.clusters();
```

Comment:
This call is not thread safe. `clusters()` should only be called on the main thread. Beyond thread safety, it's not safe since it's possible for the central cluster to get removed before the thread local cluster update goes out (if there is a quick add/remove sequence). We will need to figure out a better way to properly handle this. Can you take a look and see if you can come up with something and potentially come talk to me IRL if you want to brainstorm (I don't have any quick ideas I would need to sit and think about this)?

Reply:
Done